ci/jobs/satellite6-pulp.yaml

```diff
@@ -0,0 +1,187 @@
+
+- job-template:
+    name: "{name}"
+    node: f23-vanilla-np
+    properties:
+        - qe-ownership
+    wrappers:
+        - ansicolor
+        - ssh-agent-credentials:
+            users:
+                - '044c0620-d67e-4172-9814-dc49e443e7b6'
+        - credentials-binding:
+            - file:
+                credential-id: 7ca95790-1c85-4418-ac4c-082a366cf2db
+                variable: REMOTE_JENKINS_CREDENTIALS
+            - file:
+                credential-id: ff1c87ec-76f1-4bb1-a947-fd2114cd8fd9
+                variable: OPENSTACK_CREDENTIALS
+    builders:
+        - shell: |
+            source $OPENSTACK_CREDENTIALS
+            source $REMOTE_JENKINS_CREDENTIALS
+
+            echo "Preparing OpenStack client environment on Jenkins server..."
+            sudo yum -y update python
+            sudo yum -y install wget
+            sudo pip install pip --upgrade
+            sudo pip install virtualenv --upgrade
+            virtualenv sat-pulp
+            source sat-pulp/bin/activate
+            pip install pbr
+            sudo yum -y install redhat-rpm-config python-devel
+
+            if [ ! -f ./requirements.txt ]; then
+                echo "Downloading requirement.txt for OpenStack client..."
+                wget https://raw.githubusercontent.com/SatelliteQE/5minute/master/requirements.txt
+            fi
+            echo "Installing all dependent packages..."
+            pip install -r requirements.txt
+
+            # Only underscores, hyphens, and alphanumeric characters are allowed for keypair names
+            KEY_NAME=pulp-jenkins
+            INSTANCE_NAME=${{KEY_NAME}}-sat6-tester
+            IMAGE_ID=`glance image-list | grep '| rhel-guest-image-7.2' | awk '{{ print $2 }}'`
+            glance image-list
+
+            # Upload SSH key pair for OpenStack instance.
+            echo "Uploading the SSH key pair..."
+            if [[ -z `nova keypair-list | grep $KEY_NAME | awk '{{ print $2 }}'` ]]; then
+                echo "ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQC6DJ8fmd61DWPCMiOEuy96ajI7rL3rWu7C9NQhE9a4SfyaiBcghREHJNCz9LGJ57jtOmNV0+UEDhyvTckZI2YQeDqGCP/xO9B+5gQNlyGZ9gSmFz+68NhYQ0vRekikpb9jNdy6ZZbfZDLp1w7dxqDIKfoyu7QO3Qr3E/9CpiucQif2p+oQOVOCdKEjvGYNkYQks0jVTYNRscgmcezpfLKhqWzAre5+JaMB0kRD5Nqadm2uXKZ4cNYStrpZ4xUrnMvAqjormxW2VJNx+0716Wc2Byhg8Nva+bsOkxp/GewBWHfNPtzQGMsL7oYZPtOd/LrmyYeu/M5Uz7/6QCv4N90P pulp" >> pulp-jenkins.pub
+                echo "ssh-rsa AAAAB3NzaC1yc2EAAAABIwAAAQEAzoPajR2xtQOAfBebX69Mx9Ee4P/LMqlxQLKvF0bc79/1ayMf3IrmpY1V6JCpABvMV1830I9D9x9Tr8E9zjg2wWT14hhHsrUKSWUsy3doIwz3MtISBZPMig5AizVjH6Wl/t833zgkeHtStCYI/bmJQykj6AgB8/A4L5SRIpNnl1q7V+sw37Rmumaiqu4lRDXyTXY7mlOCuxrus/WcGyVTh2k+oBVqkz2V2s3+Or8Zy2Y441B4z3vF3lE6aoIBwidBVZ1LKaofZDMRf/lu575cI4AB3N5DQvpqwLSc4+HIvog0FdKUo3qMaFgg0KNkYS5fnpDpRDRQnFw7oFnBHiPNqw== jenkins@satellite-jenkins" >> pulp-jenkins.pub
+                nova keypair-add --pub-key pulp-jenkins.pub $KEY_NAME
+            fi
+            nova keypair-list
```

Comment:
We can assume the keypair is there, this line can be removed since the output is not needed. If for some reason the later ssh fail then manually we can inspect this.